windows/devices/uwb/UwbSession.cxx

```diff
@@ -254,6 +254,32 @@ UwbSession::GetApplicationConfigurationParametersImpl()
     }
 }
 
+UwbSessionState
+UwbSession::GetSessionStateImpl()
+{
+    uint32_t sessionId = GetId();
+
+    auto resultFuture = m_uwbDeviceConnector->SessionGetState(sessionId);
+    if (!resultFuture.valid()) {
+        PLOG_ERROR << "failed to obtain session state for session id " << sessionId;
+        throw UwbException(UwbStatusGeneric::Failed);
+    }
+
+    try {
+        auto [uwbStatus, sessionState] = resultFuture.get();
+        if (!IsUwbStatusOk(uwbStatus)) {
+            // TODO: this value should be returned
```

Comment:
The caller is expected to handle `UwbExceptions`, so just throw one with `uwbStatus` as the c'tor arg.

Reply:
I added it.